crates/red_knot_python_semantic/src/types/generics.rs

```diff
@@ -21,7 +21,7 @@ impl<'db> GenericContext<'db> {
         index: &'db SemanticIndex<'db>,
         type_params_node: &ast::TypeParams,
     ) -> Self {
-        let variables = type_params_node
+        let variables: Box<[_]> = type_params_node
```

Comment:
Haven't investigated why this is necessary. Can look into getting rid of this, if we see the need.

Reply:
The auto-generated `new` constructor for an interned struct has `Borrow`-like traits for its parameters, since it's doing a hash table lookup.  Whereas `new` for a tracked struct takes in the field type directly.